### PR TITLE
ci: Add support for FreeBSD to the build process

### PR DESCRIPTION
**Pull request by nhooyr**
Updates #1826
Updates #1880
Updates #2320
Updates #2349

### CODE REVIEW

**Comment by nhooyr** [comment]
Need to fix some other scripts rn as well.

**Comment by nhooyr** [comment]
Now closes #1826, #1880, #2349

I'm going to leave #2320 for now as it needs further investigation.

File: lib/vscode/package.json [hunk 1]
```diff
@@ -116,7 +116,6 @@
     "css-loader": "^3.2.0",
     "debounce": "^1.0.0",
     "deemon": "^1.4.0",
-    "electron": "9.3.3",
```
Comment (jsjoeio) [inline lib/vscode/package.json:119]: this was intentional, right?

**Review by jsjoeio — APPROVED** [review]
All looks good as usual! 🌮 

Only question was about removing `electron` as a package. If that was intentional, should we also remove `lib/vscode/src/typings/electron.d.ts`?

File: lib/vscode/package.json (hunk 1 above)
Comment (nhooyr) [reply lib/vscode/package.json:119]: @jsjoeio see the per commit breakdown!

**Comment by nhooyr** [comment]
@jsjoeio See commit 3

File: lib/vscode/package.json (hunk 1 above)
Comment (jsjoeio) [reply lib/vscode/package.json:119]: Ah sorry! Should have looked there first. Got it 👍🏼

**Comment by jsjoeio** [comment]
@nhooyr got it! 

Looks good! 🚀 

(side note: would be curious to hear why we should keep the types for my own learning?)

**Comment by nhooyr** [comment]
> (side note: would be curious to hear why we should keep the types for my own learning?)

VS Code won't compile without it. We still compile the electron parts for some reason. I asked @code-asher and he said the easiest way to avoid problems is just to include the types to quiet `tsc`.

**Pull request merged**